src/UniversalRewardsDistributor.sol

```diff
@@ -78,9 +78,13 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributorStaticTyping
     /// @param newIpfsHash The optional ipfs hash containing metadata about the root (e.g. the merkle tree itself).
     /// @dev Warning: The `newIpfsHash` might not correspond to the `newRoot`.
     function submitRoot(bytes32 newRoot, bytes32 newIpfsHash) external onlyUpdaterRole {
+        require(newRoot != root || newIpfsHash != ipfsHash, ErrorsLib.ALREADY_SET);
+
         if (timelock == 0) {
             _setRoot(newRoot, newIpfsHash);
         } else {
+            require(newRoot != pendingRoot.root || newIpfsHash != pendingRoot.ipfsHash, ErrorsLib.ALREADY_PENDING);
+
```

Comment:
A bit weird to check the current root (line 81), when changing only the pending root in the `else` branch. This does not correspond to the invariant "the pending root will change the current root" either, because the current root can change freely with `setRoot`. PR #100 also fixes this